src/protocol/protocol.ts

```diff
@@ -1240,3 +1248,198 @@ export namespace Session {
 
   export type UnsubscribeResult = Message.EmptyResult;
 }
+
+/** @see https://w3c.github.io/webdriver-bidi/#module-input */
+export namespace Input {
+  export type Command = PerformActions | ReleaseActions;
+
+  export type ElementOrigin = {
+    type: 'element';
+    element: CommonDataTypes.SharedReference;
+  };
+
+  export type PerformActions = {
+    method: 'input.performActions';
+    params: PerformActionsParameters;
+  };
+
+  export type PerformActionsParameters = {
+    context: CommonDataTypes.BrowsingContext;
+    actions: SourceActions[];
+  };
+
+  export type SourceActions =
+    | NoneSourceActions
+    | KeySourceActions
+    | PointerSourceActions
+    | WheelSourceActions;
+
+  export enum SourceActionsType {
+    None = 'none',
+    Key = 'key',
+    Pointer = 'pointer',
+    Wheel = 'wheel',
+  }
+
+  export type NoneSourceActions = {
+    type: SourceActionsType.None;
+    id: string;
+    actions: NoneSourceAction[];
+  };
+
+  export type NoneSourceAction = PauseAction;
+
+  export type KeySourceActions = {
+    type: SourceActionsType.Key;
+    id: string;
+    actions: KeySourceAction[];
+  };
+
+  export type KeySourceAction = PauseAction | KeyDownAction | KeyUpAction;
+
+  export type PointerSourceActions = {
+    type: SourceActionsType.Pointer;
+    id: string;
+    parameters?: PointerParameters;
+    actions: PointerSourceAction[];
+  };
+
+  export enum PointerType {
+    Mouse = 'mouse',
+    Pen = 'pen',
+    Touch = 'touch',
+  }
+
+  export type PointerParameters = {
+    /**
+     * @defaultValue `"mouse"`
+     */
+    pointerType?: PointerType;
```

Comment:
It's possible to do this in `protocol-parser.ts` by using zod's `.default()` method.